test/api/liskApi.js

```diff
@@ -122,14 +123,123 @@ describe('Lisk API module', () => {
+				LSK = new LiskAPI({ });
+				(LSK).should.have.property('ssl').be.false();
+			});
+		});
+
+		describe('on initialize randomPeer', () => {
+			it('should set randomPeer to true when no node is given and randomPeer not explicitly set', () => {
+				(LSK).should.have.property('randomPeer').be.true();
+			});
+
+			it('should set randomPeer to true on initialization with option passed', () => {
+				LSK = new LiskAPI({ randomPeer: true });
+				(LSK).should.have.property('randomPeer').be.true();
+			});
+
+			it('should set randomPeer to false on initialization with option passed', () => {
+				LSK = new LiskAPI({ randomPeer: false });
+				(LSK).should.have.property('randomPeer').be.false();
+			});
+		});
+
+		describe('on initialize port', () => {
+			it('should set port to desired port if set on initialization with option passed', () => {
+				LSK = new LiskAPI({ port: 2000 });
+				(LSK).should.have.property('port').be.equal(2000);
+			});
+
+			it('should set port to default testnet port if not set but used testnet on initialization with options passed', () => {
+				LSK = new LiskAPI({ port: undefined, testnet: true });
+				(LSK).should.have.property('port').be.equal(7000);
+			});
+
+			it('should set testnet true and port to 100 on initialization when options passed', () => {
+				LSK = new LiskAPI({ port: 100, testnet: true });
+				(LSK).should.have.property('port').be.equal(100);
+			});
+		});
+
+		describe('on initialize peers', () => {
+			it('should set all peers list to peers on initialization with option passed', () => {
+				LSK = new LiskAPI({ peers: defaultPeers });
+				(LSK).should.have.property('defaultPeers').be.equal(defaultPeers);
+				(LSK).should.have.property('defaultTestnetPeers').be.equal(defaultPeers);
+				(LSK).should.have.property('defaultSSLPeers').be.equal(defaultPeers);
+			});
+
+			it('should set all bannedPeers list to bannedPeers on initialization with option passed', () => {
+				LSK = new LiskAPI({ bannedPeers: defaultBannedPeers });
+				(LSK).should.have.property('bannedPeers').be.equal(defaultBannedPeers);
+			});
+
+			it('should set currentPeer to peer on initialization with option passed', () => {
+				LSK = new LiskAPI({ node: defaultUrl });
+				(LSK).should.have.property('currentPeer').be.equal(defaultUrl);
+			});
+		});
+
+		describe('on initialize nethash', () => {
+			it('should set nethash to devnet when own nethash used', () => {
+				const ownNethash = '123';
+				const expectedDevNethash = {
+					'Content-Type': 'application/json',
+					nethash: ownNethash,
+					broadhash: mainnetHash,
+					os: 'lisk-js-api',
+					version: '0.0.0a',
+					minVersion: '>=0.5.0',
+					port: livePort,
+				};
+				LSK = new LiskAPI({ nethash: ownNethash });
+				(LSK).should.have.property('nethash').be.eql(expectedDevNethash);
+			});
+		});
+
+		describe('interaction with config.json default values', () => {
+			it('passed in options should override default values', () => {
```

Comment:
This test assumes the `config.json` file has not been modified. If it has, this test may give false positives. Will it ever be modified?